app/interfaces/database/user_auth_repository.go

```diff
@@ -0,0 +1,143 @@
+package database
+
+//TODO: Add custom errors in way that wouldnt require importing "errors" module here (used on lines: 30, 34, 107, 113 )
+import (
+	"errors"
+
+	uuid "github.com/google/uuid"
```

Comment:
[IMO] It's not good that repository depend on the uuid package. Maybe create a uuid_handler is the better practice. But it takes long time. So  it's fine right now.

Reply:
As just discussed, uuid is just a standalone library that generates a random value based on internal info, it doesn't do any calls, any data passing etc.
So should be fine to leave as it is.